src/main/java/enmasse/kafka/bridge/tracker/SimpleOffsetTracker.java

```diff
@@ -32,52 +32,53 @@
  */
 public class SimpleOffsetTracker<K, V> implements OffsetTracker<K, V> {
 
+	private static class PartitionState {
+		public long offset;
+		public boolean changed;
+		public PartitionState(long offset, boolean changed) {
+			this.offset = offset;
+			this.changed = changed;
+		}
+	}
+	
 	// Apache Kafka topic to track
 	private String topic;	
 	// map with each partition and related tracked offset
-	private Map<Integer, Long> offsets;
-	// map with changed status of offsets
-	private Map<Integer, Boolean> offsetsFlag;
+	private Map<Integer, PartitionState> offsets;
 	
 	/**
 	 * Contructor
 	 *
 	 * @param topic	topic to track offset
 	 */
 	public SimpleOffsetTracker(String topic) {
-		
 		this.topic = topic;
 		this.offsets = new HashMap<>();
-		this.offsetsFlag = new HashMap<>();
 	}
 	
 	@Override
-	public synchronized void track(String tag, ConsumerRecord<K, V> record) {
+	public synchronized void track(int partition, long offset, ConsumerRecord<K, V> record) {
 		// nothing
 	}
 	
 	@Override
-	public synchronized void delivered(String tag) {
-		
-		// get partition and offset from delivery tag : <partition>_<offset>
-		int partition = Integer.valueOf(tag.substring(0, tag.indexOf("_")));
-		long offset = Long.valueOf(tag.substring(tag.indexOf("_") + 1));
+	public synchronized void delivered(int partition, long offset) {
 		
 		if (this.offsets.containsKey(partition)) {
 			
 			// map already contains partition but to handle "out of order" delivery
 			// we have to check that the current delivered offset is greater than
 			// the last committed offset
-			if (offset > this.offsets.get(partition)) {
-				this.offsets.put(partition, offset);
-				this.offsetsFlag.put(partition, true);
+			PartitionState x = this.offsets.get(partition);
```

Comment:
Can we use something like "state" or "partitionState" as variable name instead of just "x" ?